portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,28 +14,107 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet responsible for listing and adding comments data */
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
+  private enum Sort { NEW, OLD, NULL }
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Testing JSON");
-    messages.add("I hope this works!");
+    Sort sort = Sort.valueOf(request.getParameter("sort").toUpperCase());
+    String imgRequest = request.getParameter("img");
+
+    // Sort comments according to request parameter
+    Query query;
+    if (sort == Sort.NULL || sort == Sort.NEW) {
+      query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    } else {
+      query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    int numComments = results.countEntities(FetchOptions.Builder.withDefaults());
+
+    // Set the number of comments to print (max)
+    String quantity = request.getParameter("quantity");
+    int max = -1;
+    if (quantity != null && quantity.length() > 0 && Integer.parseInt(quantity) <= numComments) {
```

Comment:
It would be good to add a comment

// Making sure the number of comments is valid, greater than 0, and ect..